Source/WebCore/svg/SVGLengthContext.cpp

```diff
@@ -211,7 +198,12 @@ ExceptionOr<float> SVGLengthContext::convertValueFromPercentageToUserUnits(float
     if (!viewportSize)
         return Exception { ExceptionCode::NotSupportedError };
 
-    return value * dimensionForLengthMode(lengthMode, *viewportSize);
+    return convertValueFromPercentageToUserUnits(value, lengthMode, *viewportSize);
+}
+
+float SVGLengthContext::convertValueFromPercentageToUserUnits(float value, SVGLengthMode lengthMode, FloatSize viewportSize)
```

Comment:
This could just be a static function inside the file (rather than a static class method).

Reply:
Done in https://github.com/WebKit/WebKit/commit/0a6cceedab76e7444eab78de32fa00ed1deb05b7.

Reply:
@smfr - I tried to do this change and tried to see if I can do it in `.h` but it does not seems to compile. So I have undid this change as of right now. Any suggestions?

Reply:
You need to define 
```
static float convertValueFromPercentageToUserUnits(...)
```
before it is called. Just move it before SVGLengthContext::resolveRectangle() around line 45.

Reply:
OK, let me retry this locally. :-)

Reply:
@shallawa - I tried but it does not compile:

<img width="661" alt="image" src="https://github.com/user-attachments/assets/07a3fc61-ece0-4aac-80af-c674bb770e9b">

and gives error:

<img width="1051" alt="image" src="https://github.com/user-attachments/assets/72870316-cb7e-4ea2-8358-92ca794c29f3">

Reply:
I see. The error happens because there is another non static function with the same name at line 58 in the SVGContextLength.h
```
ExceptionOr<float> convertValueFromPercentageToUserUnits(float value, SVGLengthMode) const;
```
So the compiler assumes your calls are for the class method. But the class method takes two arguments and you are passing three. So it asks if you meant to call the static one `::WebCore::convertValueFromPercentageToUserUnits()`.

I guess you can leave this method as a static class method for now.

Reply:
Thanks!